src/features/reports/components/ReportTable/ReportTable.tsx

```diff
@@ -23,15 +26,24 @@ function openOSMID(
 }
 
 export function ReportTable() {
-  const [{ data, thead, ascending, sortIndex, meta }, { sortBy, setState }] =
+  const [{ data, thead, ascending, sortIndex }, { sortBy, setState }] =
     useAtom(tableAtom);
+  const [meta] = useAtom(currentReportAtom);
 
   useEffect(() => {
     return () => {
-      setState({ meta: null, sortIndex: 0, ascending: null });
+      setState({ sortIndex: 0, ascending: null });
     };
   }, []);
 
+  if (data === null) {
+    return <Text type="heading-xl">{i18n.t('No data for this report')}</Text>;
+  }
+
+  if (data === 'sorting') {
+    return <Text type="heading-xl">{i18n.t('Sorting data...')}</Text>;
```

Comment:
should we put the translations in src/core/localization/translations/en/common-messages.json with the key?
or it's ok to go this way?

Reply:
maybe we should but that's another discussion, we've mostly used this simple way that far. I've added this point to discuss on next fronted meetup